src/MappedArrays.jl

```diff
@@ -62,9 +62,10 @@ parenttype{T,N,A,F,Finv}(::Type{MappedArray{T,N,A,F,Finv}}) = A
 
 function testvalue(data)
     if !isempty(data)
-        return first(data)
-    end
-    zero(eltype(data))
+        first(data)
+    else
+        zero(eltype(data))
+    end::eltype(data)
```

Comment:
Right, it's not inferable but Base goes to some effort to generate the tightest type it can. Check out this scary function: https://github.com/JuliaLang/julia/blob/0e970f05fc9fb16b46d285f8b3fc5ff88fe53b44/base/array.jl#L446-L465

But I'm not sure we want to go there, because often for me this is all about `mmap`ped arrays and I don't really want to pay the price of a trip through the whole thing just to tighten the eltype.